src/libs/actions/Travel.ts

```diff
@@ -1,59 +1,168 @@
+Onyx.connect({
+    key: ONYXKEYS.NVP_ACTIVE_POLICY_ID,
+    callback: (val) => {
+        activePolicyID = val;
+    },
+});
+
+let primaryLogin: string;
+Onyx.connect({
+    key: ONYXKEYS.ACCOUNT,
+    callback: (val) => {
+        primaryLogin = val?.primaryLogin ?? '';
+    },
+});
+
+let isSingleNewDotEntry: boolean | undefined;
+Onyx.connect({
+    key: ONYXKEYS.IS_SINGLE_NEW_DOT_ENTRY,
+    callback: (val) => {
+        isSingleNewDotEntry = val;
+    },
+});
 
 /**
  * Accept Spotnana terms and conditions to receive a proper token used for authenticating further actions
  */
-function acceptSpotnanaTerms() {
+function acceptSpotnanaTerms(domain?: string) {
     const optimisticData: OnyxUpdate[] = [
         {
             onyxMethod: 'merge',
             key: ONYXKEYS.NVP_TRAVEL_SETTINGS,
             value: {
                 hasAcceptedTerms: true,
+            },
+        },
+        {
+            onyxMethod: 'merge',
+            key: ONYXKEYS.TRAVEL_PROVISIONING,
+            value: {
                 isLoading: true,
+                errors: null,
             },
         },
     ];
 
-    const finallyData: OnyxUpdate[] = [
+    const successData: OnyxUpdate[] = [
         {
             onyxMethod: 'merge',
-            key: ONYXKEYS.NVP_TRAVEL_SETTINGS,
+            key: ONYXKEYS.TRAVEL_PROVISIONING,
             value: {
                 isLoading: false,
             },
         },
     ];
 
-    const error = new Error('Failed to generate spotnana token.');
-
-    return new Promise((resolve, reject) => {
-        asyncOpenURL(
-            // eslint-disable-next-line rulesdir/no-api-side-effects-method
-            API.makeRequestWithSideEffects(SIDE_EFFECT_REQUEST_COMMANDS.ACCEPT_SPOTNANA_TERMS, null, {optimisticData, finallyData})
-                .then((response) => {
-                    if (!response?.spotnanaToken) {
-                        reject(error);
-                        throw error;
-                    }
-                    return buildTravelDotURL(response.spotnanaToken);
-                })
-                .catch(() => {
-                    reject(error);
-                    throw error;
-                }),
-            (travelDotURL) => {
-                resolve(travelDotURL);
-                return travelDotURL ?? '';
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: 'merge',
+            key: ONYXKEYS.TRAVEL_PROVISIONING,
+            value: {
+                isLoading: false,
+                errors: getMicroSecondOnyxErrorWithTranslationKey('travel.errorMessage'),
             },
-        );
-    });
+        },
+    ];
+
+    const params: AcceptSpotnanaTermsParams = {domain};
+
+    API.write(WRITE_COMMANDS.ACCEPT_SPOTNANA_TERMS, params, {optimisticData, successData, failureData});
+}
+
+function handleProvisioningPermissionDeniedError(domain: string) {
+    Navigation.navigate(ROUTES.TRAVEL_DOMAIN_PERMISSION_INFO.getRoute(domain));
```

Comment:
Navigation shouldn't be happening in the action files. It should be happening in the view.

Reply:
Created a follow-up [issue](https://github.com/Expensify/App/issues/55755) for this.